lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -33,6 +33,49 @@
 namespace flutter {
 
 namespace {
+
+#ifdef FML_OS_ANDROID
+class MallocDeviceBuffer : public impeller::DeviceBuffer {
+ public:
+  MallocDeviceBuffer(impeller::DeviceBufferDescriptor desc, uint8_t* data)
+      : impeller::DeviceBuffer(desc) {
+    data_ = data;
+  }
+
+  ~MallocDeviceBuffer() override { free(data_); }
+
+  bool SetLabel(const std::string& label) override { return true; }
+
+  bool SetLabel(const std::string& label, impeller::Range range) override {
+    return true;
+  }
+
+  void Flush() override {}
+
+  uint8_t* OnGetContents() const override { return data_; }
+
+  bool OnCopyHostBuffer(const uint8_t* source,
+                        impeller::Range source_range,
+                        size_t offset) override {
+    for (auto i = 0u; i < source_range.length; i++) {
+      data_[offset + i] = source[source_range.offset + i];
+    }
```

Comment:
```suggestion
    memcpy(data_ + offset), source + source_range.offset, source_range.length);
```
`memcpy` will always be as fast or faster than your own loop (usually faster).